index.js

```diff
@@ -0,0 +1,97 @@
+import { catsData } from "/data.js"
+
+const catsArray = []
+const emotionsArray = []
+const emotionRadios = document.getElementById('emotion-radios')
+const radios = document.getElementsByClassName('radio')
+const getImageBtn = document.getElementById('get-image-btn')
+const gifsOnlyCheckbox = document.getElementById('gifs-only-option')
+const memeModalInner = document.getElementById('meme-modal-inner')
+const memeModal = document.getElementById('meme-modal')
+const memeModalClose = document.getElementById('meme-modal-close-btn')
+
+emotionRadios.addEventListener('change', highlightCheckedOption)
+getImageBtn.addEventListener("click", renderCat)
+memeModalClose.addEventListener("click", closeModal)
+
+function closeModal() {
+    memeModal.style.display = "none"
+}
+
+function getMatchingCatsArray() {
+    if (document.querySelector('input[type="radio"]:checked')) {
+        const selectedEmotion = document.querySelector('input[type="radio"]:checked').value
+        const isGifChecked = gifsOnlyCheckbox.checked
+
+        const matchingCatsArray = catsData.filter(function (cat) {
+            console.log(isGifChecked)
+            if (isGifChecked) {
+                return cat.emotionTags.includes(selectedEmotion) && cat.isGif
+            }
+
+            else {
+                return cat.emotionTags.includes(selectedEmotion)
+            }
+        })
+        return matchingCatsArray
+    }
+}
+
+function getSingleCatObject() {
+    const catsArray = getMatchingCatsArray()
+    if (catsArray.length === 1) {
+        return catsArray[0]
+    }
+    else {
+        const randomIndex = Math.floor(Math.random() * catsArray.length)
+        return catsArray[randomIndex]
+    }
+}
+
+function renderCat() {
+    const catObject = getSingleCatObject()
+    memeModalInner.innerHTML = `<img 
```

Comment:
I'd use [Document.createElement](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement) and [append](https://developer.mozilla.org/en-US/docs/Web/API/Element/append) it instead. If they haven't taught you createElement, don't worry about it

Reply:
I have not learned about createElement. Should I wait to implement this?